atat/app.py

```
@@ -39,11 +39,17 @@
 from atat.utils.notification_sender import NotificationSender
 from atat.utils.session_limiter import SessionLimiter
 
-ENV = os.getenv("FLASK_ENV", "dev")
+ENV = os.getenv("FLASK_ENV", "production")
+
+# Env is development by default.
+if not (ENV in ["development", "production", "test", "ci"]):
```

I would really prefer if our allowed values were defined as an `enum.Enum` rather than "magic strings".

```python
class ApplicationEnvironment(enum.Enum):
    DEVELOPMENT = "development"
    TEST = "test"
    CI = "ci"
    PRODUCTION = "production"

ENV = ApplicationEnvironment(os.getenv("FLASK_ENV", "development"))
```

And then everywhere, we can compare to `ApplicationEnvironment.DEVELOPMENT` or `ApplicationEnvironment.PRODUCTION` or something like that.

We'd even be able to define methods or attributes as well which may be nice. But in general, if we want to enumerate allowed values, we should use an enumeration :)

---

@kylelaker-ccpo SO, if I get this well do you want this to fail instead of having a default ENV?

---

the criteria until now are to default to production

---

suggested changes 104 failed, 703 passed, 35 skipped, 1 warning in 51.70s

---

Hey guys - yes please always fail fast on things like this. Definitely never ever want to default to dev because that becomes a security risk.

---

I hold the following fairly strong opinions related to the questions raised here:

1. It is important to always fail as early and descriptively as possible. We don't necessarily know what breaks or bad (and insecure) assumptions could be made in an "invalid" environment. We should fail quickly upfront. We should let users (users means "operators" and "developers" here) know _why_ there was a failure, even if it's implicit log messages
2. Enumerations provide additional safety, easier comparisons, and greater flexibility. You also get a usable error if you try to use an invalid value.

In addition to updating the initialization of `ENV`, it is important to replace all other uses of any of the literal strings with the `Enum` name. My suggestion here cannot be applied in isolation; it is the beginnings of a larger suggestion to "use an enum." If that will require further discussion and conversation, reach out and perhaps we can do a pair programming session or quick chat on this!

---

this is no longer part of the implementation some changes were need to comply with the coverages.